ts/features/fci/components/DocumentWithSignature.tsx

```diff
@@ -0,0 +1,315 @@
+        res => res.base64()
+      );
+
+      await PDFDocument.load(pdfFromBase64(existingPdfBytes)).then(res => {
+        // get the signature field by unique name
+        pipe(
+          res.findPageForAnnotationRef(
+            res.getForm().getSignature(uniqueName).ref
+          ),
+          O.fromNullable,
+          O.map(pageRef => {
+            const page = res.getPages().indexOf(pageRef);
+            setSignaturePage(page + 1);
+            // The signature field is extracted by its unique_name.
+            // Using low-level acrofield (acrobat field) it is possible
+            // to obtain the elements of the signature field such as the
+            // box that contains it. Once the box is obtained, its
+            // coordinates are used to draw a rectangle on the related page.
+            const signature = res.getForm().getSignature(uniqueName);
+            const [widget] = signature.acroField.getWidgets();
+            const rect = widget.getRectangle();
+            res.getPage(page).drawRectangle({
+              x: rect.x,
+              y: rect.y,
+              width: rect.width,
+              height: rect.height,
+              color: rgb(0, 0.77, 0.79),
+              opacity: 0.5,
+              borderOpacity: 0.75
+            });
+          })
+        );
+
+        return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));
+      });
+    },
+    [documents, currentDoc]
+  );
+
+  /**
+   * Get the pdf url from documents,
+   * download it as base64 string and
+   * load the pdf as pdf-lib object
+   * to draw a rect over the signature field
+   * giving a set of coordinates
+   * @param attrs the signature field attrs containing the coords
+   */
+  const drawRectangleOverSignatureFieldByCoordinates = React.useCallback(
+    async (attrs: SignatureFieldToBeCreatedAttrs) => {
+      // TODO: refactor this function to use fp-ts
+      const doc = documents[currentDoc];
+      const url = doc.url;
+
+      const existingPdfBytes = await ReactNativeBlobUtil.fetch("GET", url).then(
+        res => res.base64()
+      );
+
+      await PDFDocument.load(pdfFromBase64(existingPdfBytes)).then(res => {
+        const page = attrs.page;
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        setSignaturePage(page.valueOf() + 1);
+        // The signature box is drawn using the coordinates of the signature field.
+        res.getPage(page).drawRectangle({
+          x: attrs.bottom_left.x ?? 0,
+          y: attrs.bottom_left.y ?? 0,
+          height: Math.abs(
+            (attrs.top_right.y ?? 0) - (attrs.bottom_left.y ?? 0)
+          ),
+          width: Math.abs(
+            (attrs.top_right.x ?? 0) - (attrs.bottom_left.x ?? 0)
+          ),
+          color: rgb(0, 0.77, 0.79),
+          opacity: 0.5,
+          borderOpacity: 0.75
+        });
+
+        return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));
```

Comment:
As above for line 107. I'd suggest to check `drawRectangleOverSignatureFieldByCoordinates` and `drawRectangleOverSignatureFieldById` to see if we could extract all common codes.

Reply:
@shadowsheep1  Could you give me some more information?

Reply:
I saw the return statement `return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));` is common between the two aforementioned functions like `existingPdfBytes` so I was suggesting to not duplicate code between the two functions but extract all the common parts.

Reply:
Ok thx @shadowsheep1! Yep It's correct. However, a refactor of these methods is planned to use fp-ts and remove these common parts as tracked here [https://pagopa.atlassian.net/browse/SFEQS-1601].